### ISSUE TITLE
decoder giving partial transcripts

### ISSUE THREAD

**Issue by pdcoded** [PaddlePaddle/DeepSpeech#160]
@kuke  Decoder gives partial transcripts once integrated with language model. While the greedy decoder gives full output . I used the scorer inside the swig wrapper . I suspect this is a common issue with decoder integrated with  lm. What can be the fix ??

**Comment by kuke** [comment]
@pdcoded I guess it is because the acoustic model is not well trained. In this case, the decoder cannot find an legal word and append it to the partially decoded transcripts. So please try to test on the model with full convergence.

**Comment by pdcoded** [comment]
The acoustic model is giving really good outputs when used with greedy decoder. Given that I don’t change my pb file and use this decoder at inference , how will I fix this issue ??

**Comment by pdcoded** [comment]
@kuke please see if this would help. I observed this
Actual transcript :: and she had served as a member of the indian parliament

Greedy Output :: and she had served as a member of the indian parliament

Beam Search Decoder[ with a trigram lm integrated ]: and she had served as

I deep dived to see why it is breaking :: Logits shape was (165,29)

I sliced it to logits for 1st 60 timestamps - (60,29) and fed it to the bsd again ::
Output was :: and she had served as

I did the same slicing from 65th timestamp till the end frame :: (100:29)
Output was :: member of the indian parliament

What i concluded was something breaking/stopping around the timeframe capturing the character "a". the argmax indices from the character "s" of as looks like this 19  0 28 28  1 28 28  0 28 . 19 being the index of s and couple of blanks before 1 which is the index of a.

I haven't looked deeply into the c++ implementation (https://github.com/PaddlePaddle/DeepSpeech/blob/edaed68f33b7fff2455c26b880340ba680621f5c/decoders/swig/ctc_beam_search_decoder.cpp ) but this is what I am observing purely at inference level.

**Comment by uzunovic** [comment]
I'm seeing the same problem, so I wonder if there is an update on this issue. Thanks!
